### PR TITLE
Add required-parameter-client-methods flag

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: src/JavaSettings.cs [hunk 1]
```diff
@@ -31,7 +31,25 @@ public class JavaSettings
         /// <param name="generateClientInterfaces">Whether or not interfaces will be generated for Service and Method Group clients.</param>
         /// <param name="implementationSubpackage">The sub-package that the Service and Method Group client implementation classes will be put into.</param>
         /// <param name="modelsSubpackage">The sub-package that Enums, Exceptions, and Model types will be put into.</param>
-        public JavaSettings(Action<bool> setAddCredentials, bool isAzure, bool isFluent, bool regenerateManagers, bool regeneratePom, string fileHeaderText, int maximumJavadocCommentWidth, string serviceName, string package, bool shouldGenerateXmlSerialization, bool nonNullAnnotations, bool stringDates, string clientTypePrefix, bool generateClientInterfaces, string implementationSubpackage, string modelsSubpackage)
```
Comment (RikkiGibson) [inline src/JavaSettings.cs]: These kinds of parameter lists have to fall sideways eventually :)

**Comment by daschult** [comment]
@RikkiGibson I've been picking defaults at this point by whatever the current behavior is. Later on if we decide to change the defaults, then we can let our customers know (the Fluent team).

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by daschult** [comment]
> linux-x64 restart

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Pull request merged**